subsys/bluetooth/host/gatt.c

```diff
@@ -3258,10 +3269,7 @@ static void gatt_find_type_rsp(struct bt_conn *conn, uint8_t err,
 		value.end_handle = end_handle;
 		value.uuid = params->uuid;
 
-		attr = (struct bt_gatt_attr)BT_GATT_ATTRIBUTE(&uuid_svc.uuid, 0,
-							      NULL, NULL,
-							      &value);
-		attr.handle = start_handle;
+		attr = BT_GATT_DISC_ATTRIBUTE(&uuid_svc.uuid, start_handle, &value);
```

Comment:
What do you think of inlining it like I show below? I think it may be better, but I'm not sure.
```suggestion
		attr = (struct bt_gatt_attr){
			.uuid = &uuid_svc.uuid,
			.handle = start_handle,
			.user_data = &value,
		};
```

(Omitted members of a compound literal are zero-initialized.)